src/core.js

```diff
@@ -47,6 +47,13 @@ let core = {
             "lib-version": "JS:" + core.versions.libVersion,
             "api-version": core.versions.apiVersion,
         }
+
+        if(customHeaders){
+            for (const key in customHeaders) {
+                headers[key] = customHeaders[key];
```

Comment:
In .NET this may throw exceptions if you try to add "protected" headers, like "Content-Length" or other that are supposed to be calculated by the HTTP client. Are there any similar pitfalls in JS/axios?

Reply:
In the case of "Content-Length" I found this answer on SO: 
_In Axios, If data is present it will set length calculated from data, so even if you pass header content-length, it will be overridden by code: ([Link](https://stackoverflow.com/a/59955320))_

For other [protected/forbidden headers](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name) the client will throw error if you trying to set these.
![image](https://user-images.githubusercontent.com/7857590/125373024-aba77a00-e384-11eb-98a8-9e400e074574.png)

Reply:
That error message is good enough to not worry about catching and rethrowing IMO.